nano/node/rpc.cpp

```diff
@@ -834,6 +834,31 @@ void nano::rpc_handler::block_info ()
 				block->serialize_json (contents);
 				response_l.put ("contents", contents);
 			}
+			
+			// Subtype check
+			if (block->type () == nano::block_type::state)
+			{
+				auto previous_balance (node.ledger.balance (transaction, block->previous ()));
+				if (balance < previous_balance)
+				{
+					response_l.put ("subtype", "send");
+				}
+				else
+				{
+					if (block->link ().is_zero ())
+					{
+						response_l.put ("subtype", "change");
+					}
+					else if (balance == previous_balance && !node.ledger.epoch_link.is_zero () && node.ledger.is_epoch_link (block->link ()))
```

Comment:
We don't seem to be checking if epoch_link is zero anywhere else but subtype checking - is that necessary?

Reply:
We check it in ledger.cpp processing as well